### PR TITLE
Add keyring secret store

### PR DESCRIPTION
**Pull request by ewgenius**
Added support for keyring secret store, based on top of https://docs.rs/keyring

This store is optional, and can be enabled by `keyring-secret-store` cargo feature.

To store secret requires a keychain entry named `spiced_secret_<secret_name>` with a key-value map stored as a json string and `spiced` account.

CleanShot 2024-03-11 at 12 14 00@2x

### CODE REVIEW

File: crates/runtime/Cargo.toml [hunk 1]
```diff
@@ -52,6 +52,7 @@ arrow_sql_gen = { path = "../arrow_sql_gen", optional = true }
 bb8 = {workspace = true, optional = true}
 bb8-postgres = {workspace = true, optional = true}
 base64 = "0.22.0"
+keyring = "2.3.2"
```
Comment (phillipleblanc) [inline crates/runtime/Cargo.toml]: Can we make this an optional feature?

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/Cargo.toml (hunk 1 above)
Comment (ewgenius) [reply crates/runtime/Cargo.toml]: @phillipleblanc The keyring is a cross-platform library, so I'm not sure if it's feasible to make it optional based on different criteria

Comment (phillipleblanc) [reply crates/runtime/Cargo.toml]: DuckDB is a cross-platform library, and I made that optional?

Comment (ewgenius) [reply crates/runtime/Cargo.toml]: Added `keyring-secret-store` feature to runtime

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**